darts/preprocessing/pipeline.py

```diff
@@ -0,0 +1,248 @@
+        return x
+
+    def predict(self, data: TimeSeries, *args, **kwargs) -> TimeSeries:
+        """
+        First transform data then on last transformer predict is run with transformed data.
+
+        Parameters
+        ----------
+        data
+            TimeSeries to be transformed and then predict on.
+        args
+            Positional arguments for the `predict` method
+        kwargs
+            Keyword arguments for the `predict` method
+
+        Returns
+        -------
+        TimeSeries
+            Predicted data.
+        """
+        raise_if_not(self._can_predict,
+                     "last object of pipeline doesn't have method called predict", logger)
+        x = self(data)
+        return self._transformers[-1].predict(x, *args, **kwargs)
+
+    def inverse_transform(self, data: TimeSeries) -> TimeSeries:
+        """
+        For each transformer in pipeline inverse_transform data. Then inverse transformed data is passed to next
+        transformer. Transformers are traversed in reverse order. Raises value error if not all of the transformers are
+        reversible.
+
+        Parameters
+        ----------
+        data
+            TimeSeries to be inverse transformed.
+
+        Returns
+        -------
+        TimeSeries
+            Inverse transformed data.
+        """
+        raise_if_not(self._reversible, "Not all transformers are able to perform inverse_transform", logger)
+        x = data
+        for name, transformer in zip(self._names, reversed(self._transformers)):
+            args = self._args_for_transformers[name]
+            kwargs = self._kwargs_for_transformers[name]
+            x = transformer(x, *args, inverse=True, **kwargs)
+        return x
+
+    def __getitem__(self, key: Union[str, int, slice]) -> 'Pipeline':
+        """
+        Gets subset of Pipeline based either on index, name of transformer or slice with indexes. Transformers with
+        names are picked based on the key. Resulting pipeline will deep copy transformers of original pipeline.
+
+        Parameters
+        ----------
+        key
+            Either int, str or slice.
+
+        Returns
+        -------
+        Pipeline
+            Subset of pipeline determined by key.
+        """
+        if isinstance(key, int):
+            return Pipeline([self._transformers[key]], [self._names[key]], deep=True)
+        elif isinstance(key, str):
+            raise_if_not(key in self._name_2_idx.keys(),
+                         f"No transformer named '{key}' available transformers {self._name_2_idx.keys()}", logger)
+            return Pipeline([self._transformers[self._name_2_idx[key]]], [key], deep=True)
+        elif isinstance(key, slice):
+            return Pipeline(self._transformers[key], self._names[key], deep=True)
+        raise_if(True, "Key must be int, str or slice", logger)
+
+    def __call__(self, data: TimeSeries, inverse: bool = False) -> TimeSeries:
```

Comment:
Same comment here, I don't see the advantage of this considering we would then have to specify `inverse`